zerver/lib/email_notifications.py

```diff
@@ -569,9 +569,6 @@ def handle_missedmessage_emails(
     }
 
     user_profile = get_user_profile_by_id(user_profile_id)
-    if user_profile.is_bot:  # nocoverage # TODO -- needs a test.
-        # Never email bot users.
-        return
```

Comment:
Do we want an `assert not user_profile.is_bot` here?  it's a bug if we ever get here for a bot user, and worth writing that down.

Though I suppose it's possible to have send-email events in the queue at the time the server restarts, and we wouldn't want those to cause this to error, so maybe the right model is a `logger.error` + return here?  

@alexmv for thoughts.

Reply:
I'd log an error and return, yeah.

Reply:
> Though I suppose it's possible to have send-email events in the queue at the time the server restarts, and we wouldn't want those to cause this to error, so maybe the right model is a logger.error + return here?

OK, makes sense. Added the logging code, along with a `TODO/compatibility`.